src/Microsoft.VisualStudio.Threading/JoinableTask.cs

```diff
@@ -535,6 +544,12 @@ private Task QueueNeedProcessEvent
         /// <param name="cancellationToken">A cancellation token that will exit this method before the task is completed.</param>
         public void Join(CancellationToken cancellationToken = default(CancellationToken))
         {
+            cancellationToken.ThrowIfCancellationRequested();
+            if (this.IsCompleted)
+            {
+                return;
```

Comment:
This regresses `Join()` in that it no longer throws when the task completed in a faulted state. I've added two unit tests to cover this case and will fix this method.